ostelco-ios-client/ViewControllers/EKYC/SelectIdentityVerificationMethodViewController.swift

```diff
@@ -60,23 +61,37 @@ class SelectIdentityVerificationMethodViewController: UIViewController {
         showNeedHelpActionSheet()
     }
     
-    func getMyInfoURL() -> URL? {
-        var components = URLComponents(string: Environment().configuration(PlistKey.MyInfoURL))!
-        components.queryItems = [
-            URLQueryItem(name: "client_id", value: Environment().configuration(PlistKey.MyInfoClientID)),
-            //TODO: Find the right values for the query parameters.
-            //URLQueryItem(name: "attributes", value: "name,nationality,dob,email,mobileno,regadd"),
-            URLQueryItem(name: "attributes", value: "name,sex,race,nationality,dob,email,mobileno,regadd,housingtype,hdbtype,marital,edulevel,assessableincome,ownerprivate,assessyear,cpfcontributions,cpfbalances"),
-            URLQueryItem(name: "purpose", value: "eKYC"),
-            URLQueryItem(name: "state", value: "123"),
-            URLQueryItem(name: "redirect_uri", value: Environment().configuration(PlistKey.MyInfoCallbackURL)),
-        ]
-        return components.url
+    func startMyInfoLogin() {
+        self.spinnerView = self.showSpinner(onView: self.view)
+        // Fetch the configuration from prime
+        APIManager.shared.primeAPI
+            .loadMyInfoConfig()
+            .ensure { [weak self] in
+                self?.removeSpinner(self?.spinnerView)
+                self?.spinnerView = nil
+            }
+            .done { [weak self] myInfoConfig in
+                debugPrint("MyInfoConfig.url: \(myInfoConfig.url)")
+                var components = URLComponents(string: myInfoConfig.url)!
+                // Add purpose and state parameters to the url.
+                // state parameter is currently ignored by prime.
+                components.queryItems = [
+                    URLQueryItem(name: "purpose", value: "eKYC"),
+                    URLQueryItem(name: "state", value: "123")
+                ]
+                // Show the login screen.
+                self?.showMyInfoLogin(url: components.url)
+            }
+            .catch { [weak self] error in
+                ApplicationErrors.log(error)
+                self?.showGenericError(error: error)
+            }
+
     }
     
-    func getMyInfoToken() {
-        if let url = getMyInfoURL() {
-            print("URL for API \(url.absoluteString)")
+    func showMyInfoLogin(url: URL?) {
+        if let url = url {
```

Comment:
You may want to make this a `guard let` and record an error if the URL is nil here